### PR TITLE
Fix redirects breaking list + list performance

### LINKED ISSUE #4563: Lists should follow redirects when rendering
### Evidence

image

### Relevant url?
e.g. https://openlibrary.org/people/seabelis/lists/OL140479L/Duplicate_Work?page=16 (at the bottom)

* Actual: Redirects appear broken
* Expected: Redirects should be followed, and appear as normal books

### Details

- **Logged in (Y/N)?** Y
- **Browser type/version?** FF85
- **Operating system?** Win10
- **Environment (prod/dev/local)?** prod

### Stakeholders
@seabelis

### PR DESCRIPTION
**Pull request by mekarpeles**
Closes #4563 

A patch to prevent Lists from breaking when Works are merged -- Just In Time resolves Work redirects to the correct record.

### Stakeholders

@cdrini @seabelis

### CODE REVIEW

File: openlibrary/templates/type/list/view.html [hunk 1]
```diff
@@ -267,22 +275,26 @@ <h6 class="collapse black uppercase">$label</h6>
                     </div>
                 </div>
 
-        $ subjects = list.get_subjects()
+        $if False:
```
Comment (cclauss) [inline openlibrary/templates/type/list/view.html:278]: ?

File: openlibrary/templates/type/list/view.html [hunk 2]
```diff
@@ -2,8 +2,12 @@
 
 $ title = list.name
 $ title_with_site = _("%(name)s | Lists | Open Library", name=title)
+
+$ component_times = {}
+$ component_times['TotalTime'] = time()
+
 $ page = safeint(query_param('page'), 1) - 1
-$ page_size = 50
+$ page_size = 25
```
Comment (mekarpeles) [inline openlibrary/templates/type/list/view.html:10]:
```suggestion
$ page_size = 50
```

File: openlibrary/core/lists/model.py [hunk 3]
```diff
@@ -358,35 +367,9 @@ def get_solrdata(self):
...
-                return {
-                    'ebook_count': sum(doc.get('ebook_count_i', 0) for doc in result.docs),
-                    'edition_count': sum(doc.get('edition_count', 0) for doc in result.docs),
-                    'work_count': result.num_found,
-                    'last_update': last_update
-                }
-        return {}
+        return {
+            'last_update': self.document.get('last_modified')
+        }
```
Comment (cdrini) [inline openlibrary/core/lists/model.py:372]: TODO: Let's create a new issue to remove this entirely.

**Review by cdrini — APPROVED** [review]
Tested on call:

- Lists with subjects still work
- Lists with authors still work
- Lists with works/editions work
- Lists with work redirects no longer display broken links/covers
- Lists no longer make a bunch of solr requests and are now much faster!

**Comment by cdrini** [comment]
Ah, missed this little bug (0 editions):

image

**Pull request merged**